lib/Dialect/TritonGPU/Transforms/FuseNestedLoops.cpp

```diff
@@ -1006,6 +1006,9 @@ static LogicalResult speculateInnerLoopLength(scf::ForOp outerLoop,
   newInnerLoop.replaceAllUsesWith(newInnerLoop.getInits());
   newInnerLoop.erase();
 
+  // Clear up the warp specialization attributes for the specialized loop.
+  newLoop->removeAttr(kWarpSpecializeAttrName);
```

Comment:
@Mogball, not 100% sure it makes sense but otherwise the specialized loops get warp specialized and may use extra smem